Documentation/ceph-object-store-crd.md

```diff
@@ -92,12 +92,18 @@ When the `zone` section is set pools with the object stores name will not be cre
 The gateway settings correspond to the RGW daemon settings.
 
 * `type`: `S3` is supported
-* `sslCertificateRef`: If the certificate is not specified, SSL will not be configured. If specified, this is the name of the Kubernetes secret that contains the SSL certificate to be used for secure connections to the object store. Rook will look in the secret provided at the `cert` key name. The value of the `cert` key must be in the format expected by the [RGW service](https://docs.ceph.com/docs/master/install/ceph-deploy/install-ceph-gateway/#using-ssl-with-civetweb): "The server key, server certificate, and any other CA or intermediate certificates be supplied in one file. Each of these items must be in pem form."
+* `sslCertificateRef`: If specified, this is the name of the Kubernetes secret that contains the SSL certificate to be used for secure connections to the object store. Rook will look in the secret provided at the `cert` key name. The value of the `cert` key must be in the format expected by the [RGW service](https://docs.ceph.com/docs/master/install/ceph-deploy/install-ceph-gateway/#using-ssl-with-civetweb): "The server key, server certificate, and any other CA or intermediate certificates be supplied in one file. Each of these items must be in pem form."
 * `port`: The port on which the Object service will be reachable. If host networking is enabled, the RGW daemons will also listen on that port. If running on SDN, the RGW daemon listening port will be 8080 internally.
-* `securePort`: The secure port on which RGW pods will be listening. An SSL certificate must be specified.
+* `securePort`: The secure port on which RGW pods will be listening. An SSL certificate must be specified either via `sslCerticateRef` or `annotations`
 * `instances`: The number of pods that will be started to load balance this object store.
 * `externalRgwEndpoints`: A list of IP addresses to connect to external existing Rados Gateways (works with external mode). This setting will be ignored if the `CephCluster` does not have `external` spec enabled. Refer to the [external cluster section](ceph-cluster-crd.md#external-cluster) for more details.
-* `annotations`: Key value pair list of annotations to add.
+* `annotations`: Key value pair list of annotations to add. The TLS secrets can be passsed to RGW with following:
+```yaml
+gateway:
+  annotations:
```

Comment:
Actually the annotations are used for something else already, these annotations will be added to all the rgw pods. Since the new feature has a very different purpose to apply to the service, let's add a new property to the gateway, such as `servingCertSecretName`

Reply:
@travisn : Is it harmful to have this part of `annotations`? Having it in`annotations` will match with the how `service serving cert`  consumed. I agree this change need to be defined only in `object-openshift.yaml`

Reply:
If it remains on these annotations, you will have extra annotations on the rgw pods, and also extra annotations on the rgw service. While extra annotations don't hurt, they have a distinct purpose so I'm suggesting we keep them separate. The rgw pods would get their annotations from this CR property, and the rgw service would get the annotation from the new property, only for the cert. Instead of my other name proposal, perhaps we call it `serviceAnnotations` so other arbitrary annotations can also be added to the service if desired.